File: 125.py
# // correct code
def ispalindrome(s):
    res = ""
    elle = "abcdefghijklmnopqrstuvwxyz1234567890"
    for elem in s.lower():
        if elem in elle:
            res+=elem
    left = 0
    right =len(res)-1
    while(left<right):
        if res[left]!=res[right]:
            return False
        else:
            left+=1
            right-=1
    return True
s="1b1"
if(ispalindrome(s)):
    print("true")
else:
    print("false")

File: 125.cpp
// done 
#include<bits/stdc++.h>
#include<cctype>
#include<iostream>

using namespace std;

bool isPalindrome(string s){
  string res="";
  string alpha = "abcdefghijklmnopqrstuvwxyz1234567890";
  for(auto it: s){
    if(isalpha(it) || isdigit(it)){
      res+=it;
    }
  }
  int left = 0;
  int right = s.length()-1;
  while(left<right){
    if(res[left]!=res[right]){
      return false;
    }else{
      left++;
      right--;
    }
  }
  return true;
}

int main(){
  string s="yo : banana ,  boy";
  if(isPalindrome(s)){
    cout<<"true";
  }else{
    cout<<"false";
  }
  return 0;
}
